applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_viscoplasticity_3d.cpp

```diff
@@ -0,0 +1,226 @@
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Alejandro Cornejo & Lucia Barbu
+//  Collaborator:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/math_utils.h"
+#include "structural_mechanics_application_variables.h"
+#include "custom_constitutive/small_strain_isotropic_plasticity_factory_3d.h"
+#include "custom_constitutive/viscous_generalized_maxwell_3d.h"
+#include "generic_small_strain_viscoplasticity_3d.h"
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer GenericSmallStrainViscoplasticity3D::Create(Kratos::Parameters NewParameters) const
+{
+    ConstitutiveLaw::Pointer p_plasticity_cl = SmallStrainIsotropicPlasticityFactory3D().Create(NewParameters);
+    ConstitutiveLaw::Pointer p_viscous_cl = Kratos::make_shared<ViscousGeneralizedMaxwell3D>();
+    return Kratos::make_shared<GenericSmallStrainViscoplasticity3D>(p_plasticity_cl, p_viscous_cl);
+
+    //return GenericSmallStrainViscoplasticity3D(p_plasticity_cl, p_viscous_cl).Clone();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void GenericSmallStrainViscoplasticity3D::CalculateMaterialResponsePK1(ConstitutiveLaw::Parameters &rValues)
+{
+    this->CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void GenericSmallStrainViscoplasticity3D::CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters &rValues)
+{
+    this->CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void GenericSmallStrainViscoplasticity3D::CalculateMaterialResponseKirchhoff(ConstitutiveLaw::Parameters &rValues)
+{
+    this->CalculateMaterialResponseCauchy(rValues);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void GenericSmallStrainViscoplasticity3D::CalculateMaterialResponseCauchy(ConstitutiveLaw::Parameters &rValues)
+{
+    ConstitutiveLaw::Pointer plaw = this->GetPlasticityConstitutiveLaw();
+
+    Vector plastic_strain = ZeroVector(6);
+
+    plaw->GetValue(PLASTIC_STRAIN_VECTOR, plastic_strain);
+
+    mpPlasticityConstitutiveLaw->GetValue(PLASTIC_STRAIN_VECTOR, plastic_strain);
+    //KRATOS_WATCH(plastic_strain)
+
+    //Vector& strain_vector = rValues.GetStrainVector();
```

Comment:
leftover comments

Reply:
this CL is work in progress yet...

Reply:
imo this should not be in master then

Wont block bcs of this